### PR TITLE
[WIP] Attempt at fixing appveyor

### PR DESCRIPTION
**Pull request by johnhaley81**
The clone tests are breaking appveyor and this PR is my attempt to figure out why since I can't consistently repo the test fails locally.

### CODE REVIEW

File: test/tests/clone.js [hunk 1]
```diff
@@ -3,8 +3,9 @@ var path = require("path");
 var Promise = require("nodegit-promise");
 var promisify = require("promisify-node");
 var fse = promisify(require("fs-extra"));
+var fixAppveyor = process.env.APPVEYOR ? describe.skip : describe;
 
-describe("Clone", function() {
+fixAppveyor("Clone", function() {
```
Comment (tbranyen) [inline test/tests/clone.js]: fix is probably not the right word here :stuck_out_tongue_winking_eye:

**Comment by johnhaley81** [comment]
So something is legit killing the the build for node v0.11.14 on windows. Which is awesome since it was [working fine yesterday](https://ci.appveyor.com/project/TimBranyen/nodegit/build/850).

I've started to narrow it down. The build is fine in v0.11.13 but will break in v0.11.14 on windows only. I don't know if a dev dependency got updated (or needs to be updated) for this to work in v0.11.14 but I'm still looking into it.

**Comment by johnhaley81** [comment]
It seems like a bug in node-gyp when running `node-gyp configure` it tries to download a file (node.lib) and has a non-matching checksum.

I get this output from the console

```
C:\git-projects\nodegit (master)                                                                                                                                                                     
λ node-gyp configure                                                                                                                                                                                 
gyp info it worked if it ends with ok                                                                                                                                                                
gyp info using node-gyp@1.0.2                                                                                                                                                                        
gyp info using node@0.11.14 | win32 | x64                                                                                                                                                            
gyp http GET http://nodejs.org/dist/v0.11.14/node-v0.11.14.tar.gz                                                                                                                                    
gyp http 200 http://nodejs.org/dist/v0.11.14/node-v0.11.14.tar.gz                                                                                                                                    
gyp http GET http://nodejs.org/dist/v0.11.14/SHASUMS256.txt                                                                                                                                          
gyp http GET http://nodejs.org/dist/v0.11.14/x64/node.lib                                                                                                                                            
gyp http GET http://nodejs.org/dist/v0.11.14/node.lib                                                                                                                                                
gyp http 200 http://nodejs.org/dist/v0.11.14/SHASUMS256.txt                                                                                                                                          
gyp http 200 http://nodejs.org/dist/v0.11.14/x64/node.lib                                                                                                                                            
gyp http 200 http://nodejs.org/dist/v0.11.14/node.lib                                                                                                                                                
gyp WARN install got an error, rolling back install                                                                                                                                                  
gyp ERR! configure error                                                                                                                                                                             
gyp ERR! stack Error: x64/node.lib local checksum 8b539aaaa72b00aa6b92b2dc9ee377159eaed4f01d0b7264a3c8ce06ac37971b not match remote 1c93d92da38a85a36b1e93556d16024d7ca6d342f9bfb9aafe066427a35c4928 
gyp ERR! stack     at deref (C:\Users\johnh\AppData\Roaming\npm\node_modules\node-gyp\lib\install.js:299:20)                                                                                         
gyp ERR! stack     at Request.<anonymous> (C:\Users\johnh\AppData\Roaming\npm\node_modules\node-gyp\lib\install.js:383:24)                                                                           
gyp ERR! stack     at Request.emit (events.js:129:20)                                                                                                                                                
gyp ERR! stack     at IncomingMessage.<anonymous> (C:\Users\johnh\AppData\Roaming\npm\node_modules\node-gyp\node_modules\request\request.js:1266:12)                                                 
gyp ERR! stack     at IncomingMessage.emit (events.js:129:20)                                                                                                                                        
gyp ERR! stack     at _stream_readable.js:907:16                                                                                                                                                     
gyp ERR! stack     at process._tickCallback (node.js:372:11)                                                                                                                                         
gyp ERR! System Windows_NT 6.3.9600                                                                                                                                                                  
gyp ERR! command "node" "C:\\Users\\johnh\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure"                                                                              
gyp ERR! cwd C:\git-projects\nodegit                                                                                                                                                                 
gyp ERR! node -v v0.11.14                                                                                                                                                                            
gyp ERR! node-gyp -v v1.0.2                                                                                                                                                                          
gyp ERR! not ok                                                                                                                                                                                      
                                                                                                                                                                                                     
C:\git-projects\nodegit (master)                                                                                                                                                                     
λ                                                                                                                                                                                                    
```

Don't really know why it's doing this. When running this in v0.11.13 I get the following output:

```
C:\git-projects\nodegit (master)
λ node-gyp configure
gyp info it worked if it ends with ok
gyp info using node-gyp@1.0.2
gyp info using node@0.11.13 | win32 | x64
gyp info spawn python
gyp info spawn args [ 'C:\\Users\\johnh\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'msvs',
gyp info spawn args   '-G',
gyp info spawn args   'msvs_version=auto',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\git-projects\\nodegit\\build\\config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\johnh\\AppData\\Roaming\\npm\\node_modules\\node-gyp\\addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   'C:\\Users\\johnh\\.node-gyp\\0.11.13\\common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=C:\\Users\\johnh\\.node-gyp\\0.11.13',
gyp info spawn args   '-Dmodule_root_dir=C:\\git-projects\\nodegit',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'C:\\git-projects\\nodegit\\build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info ok

C:\git-projects\nodegit (master)
λ
```

Which is what I was expecting. Similar results also on node v0.10.*.

**Comment by tbranyen** [comment]
Node 0.12 is vaporware at this point should we just ignore it?

**Comment by johnhaley81** [comment]
yeah, we can ignore node 0.12

**Comment by maxkorp** [comment]
Well, are we dropping support for node 0.11? I mean it's meant to be unstable but that seems like a big change.

**Comment by mcollina** [comment]
If we support io.js then ok for not supporting 0.11.x. There should be at some point a 0.11.15 for general review which will be 0.12 (kind of). But at this point is all nonsense.

**Comment by maxkorp** [comment]
I agree. 0.12 _is_ pretty much vaporware. IOjs dev sort of jumpstarted it back up a bit, but not really. Still missing deadlines by weeks/months now. if we can get it running consistantly with IO then i'm fine ditching 11, since IO is the new unstable, essentially.

**Comment by johnhaley81** [comment]
Alright, I'm going to fix this guy up then and merge it in.

**Comment by johnhaley81** [comment]
I'll also keep an eye on it and try to figure out the underlying issue on whatever spare time I have (lolololol).

If I can figure out wtf is happening and why it's dying then I'll put it back in. I wouldn't hold my breath on that though.

**Pull request merged**